public/coffee/ide.coffee

```diff
@@ -127,6 +127,13 @@ define [
 		ide.permissionsManager = new PermissionsManager(ide, $scope)
 		ide.binaryFilesManager = new BinaryFilesManager(ide, $scope)
 
+		# Set up labels
+		$scope.$on 'doc:labels:updated', labels.onDocLabelsUpdated
+		$scope.$on 'entity:deleted', labels.onEntityDeleted
+		$scope.$on 'file:upload:complete', labels.fileUploadComplete
+		$timeout () ->
+			labels.loadProjectLabelsFromServer()
+
```

Comment:
Because `LabelsManager` is part of the Ace directive, it get's initialized more than once per page load. It feels cleaner to have the ide-global behaviours be init'd in the ide.